skfp/preprocessing/filters/rule_of_3.py

```diff
@@ -0,0 +1,112 @@
+from typing import Optional
+
+from rdkit.Chem import Mol
+from rdkit.Chem.Crippen import MolLogP
+from rdkit.Chem.Descriptors import MolWt
+from rdkit.Chem.rdMolDescriptors import (
+    CalcNumHBA,
+    CalcNumHBD,
+    CalcNumRotatableBonds,
+    CalcTPSA,
+)
+
+from skfp.bases.base_filter import BaseFilter
+
+
+class RuleOf3(BaseFilter):
+    """
+    Rule optimised to search for fragment-based lead-like compounds with desired properties.
+    It was described in [1]
+
+    Acording to this rule fragment-based lead-like compound should meet following criteria:
+    - molecular weight <= 300 daltons
+    - HBA <= 3
+    - HBD <= 3
+    - logP <= 3
+    - TPSA <= 60 (only in extemded version of this rule)
+    - number of rotatable bonds <= 3 (only in extemded version of this rule)
+    Parameters
```

Comment:
Empty line between description and `Parameters`

Reply:
Done